common/node/nns_ros_bridge.cc

```diff
@@ -0,0 +1,172 @@
+ *
+ * @bug     No known bugs.
+ */
+#include <ros/ros.h>
+#include <std_msgs/MultiArrayLayout.h>
+#include <std_msgs/MultiArrayDimension.h>
+
+#include "nns_ros_bridge.h"
+#include "nns_ros_bridge/tensors.h"
+
+const char BASE_NODE_NAME[] = "tensor_ros_sink";
+const guint DEFAULT_Q_SIZE = 1000;
+
+/**
+ * @brief	The only public constructor of this class
+ * @param[in] node_name : The name of the ROS node under /tensor_ros_sink
+ * @param[in]  topic_name : The name of topic what this class concerns
+ * @return None
+ */
+NnsRosBridge::NnsRosBridge (const char *node_name, const char *topic_name)
+{
+  /* ROS initialization requires command-line arguments */
+  int dummy_argc = 0;
+  char **dummy_argv = NULL;
+
+  this->str_node_name = std::string (node_name);
+  this->str_pub_topic_name = std::string (topic_name);
+
+  ros::init (dummy_argc, dummy_argv, this->str_node_name);
+  this->nh_parent = new ros::NodeHandle (BASE_NODE_NAME);
+  this->nh_child = new ros::NodeHandle (*(this->nh_parent),
+      this->str_node_name);
+
+  this->ready_to_pub = FALSE;
+  this->nns_tensors_info_pub = NULL;
+
+  /**
+   * The name of published topic would be
+   * /tensor_ros_sink/PID_${PID}/${ElementNameOfTensorRosSink}
+   */
+  this->ros_sink_pub = this->nh_child->advertise<nns_ros_bridge::tensors>(
+      this->str_pub_topic_name, DEFAULT_Q_SIZE);
+}
+
+/**
+ * @brief	A method for configuraing topic information to publish
+ * @param[in] conf : The configuration information at the NNStreamer side
+ * @return TRUE if the configuration is successfully done
+ */
+gboolean NnsRosBridge::setPubTopicInfo (const GstTensorsConfig *conf)
+{
+  const GstTensorsInfo *tensors_info = &conf->info;
+
+  g_return_val_if_fail (tensors_info->num_tensors <= NNS_TENSOR_SIZE_LIMIT,
+    FALSE);
+
+  this->num_of_tensors_pub = tensors_info->num_tensors;
+
+  for (guint i = 0; i < this->num_of_tensors_pub; ++i) {
+    gsize tensor_size = gst_tensor_info_get_size (&tensors_info->info[i]);
+
+    for (guint j = 0; j < NNS_TENSOR_RANK_LIMIT; ++j) {
+      std_msgs::MultiArrayDimension each_dim;
+      /**
+       * FIXME: Since the type of 'stride' of std_msgs::MultiArrayDimension is
+       * unsigned int,conversion from size_t to unsigned int is requried.
+       */
+      guint each_dim_stride = 1;
+
+      for (guint k = 0; k <= NNS_TENSOR_RANK_LIMIT - (j + 1); ++k) {
+        each_dim_stride *= tensors_info->info[i].dimension[k];
+      }
+      each_dim.stride = each_dim_stride;
+      each_dim.label = tensor_element_typename[tensors_info->info[i].type];
+      each_dim.size =
+          tensors_info->info[i].dimension[NNS_TENSOR_RANK_LIMIT - (j + 1)];
+
+      this->topic_layouts_pub[i].dim.push_back (each_dim);
+    }
+
+    g_return_val_if_fail (
+        this->topic_layouts_pub[i].dim.at(0).stride == tensor_size, FALSE
+        );
+  }
+
+  this->ready_to_pub = TRUE;
+  this->nns_tensors_info_pub = (GstTensorsInfo *) &(conf->info);
```

Comment:
Question : nns_tensors_info_pub is it necessary?
IMO it is better to copy the tensor info (conf->info) into nns_tensors_info_pub.

Reply:
A: nns_tensors_info_pub is unnecessary now. Removed.